scapy/layers/l2.py

```diff
@@ -728,17 +793,22 @@ class ARP_am(AnsweringMachine):
     send_function = staticmethod(sendp)
 
     def parse_options(self, IP_addr=None, ARP_addr=None):
+        # type: (Optional[str], Optional[str]) -> None
         self.IP_addr = IP_addr
         self.ARP_addr = ARP_addr
 
     def is_request(self, req):
-        return (req.haslayer(ARP) and
-                req.getlayer(ARP).op == 1 and
-                (self.IP_addr is None or self.IP_addr == req.getlayer(ARP).pdst))  # noqa: E501
+        # type: (Ether) -> bool
+        if not req.haslayer(ARP):
+            return False
+        arp = cast(ARP, req.getlayer(ARP))
```

Comment:
You can use `arp = req[ARP]` instead of `getlayer` to avoid having to cast it (using [] returns an exception if the layer is minssing, whereas `getlayer` returns None for historical reasons)